errors.go

```diff
@@ -27,6 +27,10 @@ var (
 	// progress because there is no change in location after Linesearcher step
 	// due to floating-point arithmetic.
 	ErrNoProgress = errors.New("linesearch: no change in location after Linesearcher step")
+
+	// ErrMaximumStep signifies that a Linesearcher reached a maximum allowed
+	// step that is not satisfactory.
+	ErrMaximumStep = errors.New("linesearch: reached maximum step")
```

Comment:
Changed to `ErrLinesearcherBound`. At the moment I will leave it as an ordinary error but I can imagine that storing the step and the bounds could be potentially useful.